airbyte-integrations/connectors/source-bigquery/unit_tests/integration/test_read_full_refresh.py

```diff
@@ -0,0 +1,117 @@
+from integration.config import ConfigBuilder
+from integration.bq_query_builder import build_query
+from integration.request_builder import BigqueryRequestBuilder
+from integration.response_builder import BigqueryResponseBuilder
+from source_bigquery import SourceBigquery
+
+# _STREAM_NAME = "events"
+_NOW = datetime.now(timezone.utc)
+# _A_START_DATE = _NOW - timedelta(days=60)
+# _ACCOUNT_ID = "account_id"
+# _CLIENT_SECRET = "client_secret"
+_NO_STATE = StateBuilder().build()
+_NO_CATALOG = CatalogBuilder().build()
+# _AVOIDING_INCLUSIVE_BOUNDARIES = timedelta(seconds=1)
+# _SECOND_REQUEST = timedelta(seconds=1)
+# _THIRD_REQUEST = timedelta(seconds=2)
+
+
+
+
+@freezegun.freeze_time(_NOW.isoformat())
+class ReadFullRefreshTest(TestCase):
+
+    @HttpMocker()
+    def test_basic(self, http_mocker: HttpMocker) -> None:
+        config = ConfigBuilder().default().build()
+
+        dataset_id = "dataset_id_1"
+        table_id = "table_id_1"
+
+        http_mocker.get(
+            BigqueryRequestBuilder.datasets_endpoint(project_id=config["project_id"]).build(),
+            BigqueryResponseBuilder.datasets(dataset_ids=[dataset_id]).build()
+        )
+
+        http_mocker.get(
+            BigqueryRequestBuilder.tables_endpoint(project_id=config["project_id"], dataset_id=dataset_id).build(),
+            BigqueryResponseBuilder.tables(table_ids=[table_id]).build()
+        )
+
+        http_mocker.post(
+            BigqueryRequestBuilder.queries_endpoint(project_id=config["project_id"]).with_body(
+                build_query(
+                    dataset_id=dataset_id,
+                    table_id=table_id,
+                    timeout_ms=30000,
+                    max_results=10000,
+                )
+            ).build(),
+            BigqueryResponseBuilder.queries().build()
+        )
+
+        http_mocker.post(
+            BigqueryRequestBuilder.queries_endpoint(project_id=config["project_id"]).with_body(
+                build_query(
+                    dataset_id=dataset_id,
+                    table_id=table_id,
+                    dry_run=True,
+                )
+            ).build(),
+            BigqueryResponseBuilder.queries().build()
+        )
+
+        http_mocker.post(
+            BigqueryRequestBuilder.queries_endpoint(project_id=config["project_id"]).with_body(
+                build_query(
+                    project_id=config["project_id"],
+                    dataset_id=dataset_id,
+                    table_id="INFORMATION_SCHEMA.KEY_COLUMN_USAGE",
+                    where=f"table_name='{table_id}'",
+                    query_end_char=";",
+                    timeout_ms=30000,
+                )
+            ).build(),
+            BigqueryResponseBuilder.query_information_schema().build()
+        )
+
+        catalog = CatalogBuilder().with_stream(
+            ConfiguredAirbyteStreamBuilder().with_name(
+                f"{dataset_id}.{table_id}"
+            ).with_primary_key(
```

Comment:
Why is it important to mention that the primary_key is none?

Reply:
Because ConfiguredAirbyteStreamBuilder sets it to [["id"]] by default and I don't want a primary key.